ui_testing/testcases/header_tests/test08_usermanagement.py

```diff
@@ -238,4 +221,260 @@ def test010_update_user_role_with_save_cancel_btn(self, save):
+        random_user_email = self.base_page.generate_random_email()
+        self.header_page.create_new_user(user_name=random_user_name, user_email=random_user_email,
+                                         user_role='Admin', user_password='1', user_confirm_password='1')
+
+        self.base_selenium.LOGGER.info(
+            'search to make sure that the role created '.format(random_user_name))
+        created_user = self.header_page.search(random_user_name)[0]
+        user_data = self.base_selenium.get_row_cells_dict_related_to_header(row=created_user)
+        self.assertTrue(created_user, user_data)
+
+        # create role with the same name
+        self.header_page.create_new_user(user_name=random_user_name, user_email=random_user_email,
+                                         user_role='Admin', user_password='1', user_confirm_password='1')
+        self.base_selenium.LOGGER.info(
+            'waiting fo validation message appear when I enter two users with the same name')
+        validation_result = self.base_selenium.wait_element(element='general:oh_snap_msg')
+
+        self.base_selenium.LOGGER.info(
+            'Assert the error message to make sure that validation when I enter two users with the same name? {}'.format(
+                validation_result))
+        self.assertTrue(validation_result)
+
+
+class LoginRandomUser(BaseTest):
+
+    def setUp(self):
+        super().setUp()
+        random_user_name = self.generate_random_string()
+        random_user_email = self.base_page.generate_random_email()
+        random_user_password = self.generate_random_string()
+        self.users_api.create_new_user(random_user_name, random_user_email, random_user_password)
+        self.login_page.login(username=random_user_name, password=random_user_password)
+        self.base_selenium.wait_until_page_url_has(text='dashboard')
+        self.header_page.click_on_header_button()
+
+    def test018_delete_user_used_in_other_entity(self):
+        """
+        User management: Make sure that you can't delete any user record If this record used in other entity
+        LIMS-6407
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+
+        last_row = self.header_page.get_last_user_row()
+        self.header_page.click_check_box(source=last_row)
+
+        self.header_page.archive_entity(menu_element='user_management:right_menu',
+                                        archive_element='user_management:archive')
+        self.header_page.get_archived_entities(menu_element='user_management:right_menu',
+                                               archived_element='user_management:archived')
+
+        last_row = self.header_page.get_last_user_row()
+        self.header_page.click_check_box(source=last_row)
+        self.header_page.delete_entity()
+        self.assertTrue(self.base_selenium.element_is_displayed(element='general:confirmation_pop_up'))
```

Comment:
- Test case description says that _Make sure that you can't delete any user record If this record used in other entity_ whoever the test case implementation delete the new user. Why?

Reply:
@0xIslamTaha this test case should be implemented by create new user you will find this in steps in the test case itself

Reply:
I can't see how this code matching those steps:
![image](https://user-images.githubusercontent.com/17981089/73726578-1ab08600-4730-11ea-8b5c-3b510143972a.png)

Reply:
@0xIslamTaha this test case was incorrect on Jira so I updated it